packages/contracts/contracts/L2/L1Block.sol

```diff
@@ -16,12 +16,14 @@ contract L1Block {
     uint256 public timestamp;
     uint256 public basefee;
     bytes32 public hash;
+    uint256 public seqnr;
```

Comment:
number, timestamp and seqnr could all be `uint64` instead of `uint256`. This would also help fix a concern from @tynes I think. Anyone opposed to changing the types?

Reply:
I'm into changing the types to `uint64`s and also into tightly packing the calldata but consider that a nice to have, would require a bunch of changes in tests to do so. If we really want to optimize then we don't even need a function selector to be honest and just use a `fallback` function + a solidity lib that can build the calldata easily for the tests

Reply:
TBH I wouldn't mind handwritten assembly for this contract. It seems like a lot of the solidity stuff (plus the ABI) makes the bytecode much longer than it needs to be.

Reply:
Going for `uint64` for now. We can do the fancy calldata packing and fallback function thing at a later stage if we really want to. And if we're planning to add more logic to this contract (like the history accumulator), then I do like to maintain the ability to also write some regular solidity in the contract. I guess we can combine it though.